### PR TITLE
Manpage fixes

### LINKED ISSUE #959: parameterize setup program name in manpages
See https://github.com/fabiangreffrath/crispy-doom/issues/235

I plan to address this, but if/when I start on it I'll assign myself this issue, so anyone who might want to work on it please feel free if I haven't done that yet.

### PR DESCRIPTION
**Pull request by jmtd**
FIxes #959.

Overhaul documentation to parameterize PROGRAM_PREFIX (e.g. chocolate-), PACKAGE_NAME (e.g. Chocolate) as well as various other existing problems like the non-doom default cfg manpages referencing themselves as default.cfg, the chocolate-foo-setup manpages referencing chocolate-setup, etc.

- [x] parameterize PROGRAM_PREFIX and PACKAGE_NAME in all manpage names
- [x] parameterize PROGRAM_PREFIX and PACKAGE_NAME in all manpage texts
- [x] more parameterization/expansion in INSTALL.*
- [x] heretic/hexen/strife.cfg reference default.cfg/doom
- [x] likewise -{heretic,hexen,strife}.cfg (default.cfg.template)
- [x] remove chocolate-setup manpage (keep foo-bar-setup)
- [x] *-foo-setup is referenced as just foo-setup in the text
- [x] <s>don't build chocolate-setup</s> it's pragmatic to build it as an intermediary for the final binaries and we already don't install it

### CODE REVIEW

**Comment by jmtd** [comment]
I think we should deprecate `chocolate-setup` and only install `chocolate-*-setup`.  That way `chocolate-heretic-setup.6` (etc.) can correctly refer to their binary name (instead of `chocolate-setup`) in the text.

**Comment by jmtd** [comment]
By deprecate I mean remove outright...

File: man/doom.template [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
-chocolate\-doom \- historically compatible Doom engine
+@PROGRAM_SPREFIX@\-doom \- historically compatible Doom engine
 .SH SYNOPSIS
-.B chocolate\-doom
+.B @PROGRAM_SPREFIX@\-doom
 [\fIOPTIONS\fR]
 .SH DESCRIPTION
 .PP
-Chocolate Doom is a port of Id Software's 1993 game "Doom" that is designed
+@PACKAGE_NAME@ is a port of Id Software's 1993 game "Doom" that is designed
```
[the remark this answers is not in the document]
Comment (jmtd) [reply man/doom.template]: You're right that we can only guess what a fork is going to do, and if we  guess wrong, they can just go ahead and diverge in the fork. Thanks for this suggestion; I intended to be consistent betwen doom.template, heretic.template etc., and I missed this one.

File: man/docgen [hunk 2]
```diff
@@ -456,37 +458,40 @@ def manpage_output(targets, template_file):
...
-    print_template(template_file, content)
+    print_template(template_file, substs, content)
 
 def usage():
-    print("Usage: %s [-V] [-c tag] [-g game] ( -m | -w | -p ) <dir>..." \
+    print("Usage: %s [-V] [-c tag] [-g game] -n program_name -s package_name [ -z shortname ] ( -m | -w | -p ) <dir>..." \
             % sys.argv[0])
     print("   -c :  Provide documentation for the specified configuration file")
     print("         (matches the given tag name in the source file)")
+    print("   -s :  Package name (for substitution)")
```
[the remark this answers is not in the document]
Comment (jmtd) [reply man/docgen]: Good idea

**Comment by jmtd** [comment]
> Apart from the few nit-picks I commented last week this looks pretty good to me.

Thanks, and thanks for the comments - I've implemented the two changes I can find in this PR, but I still can't find the nits you are referring to here. If they are worth it please give me a clue where to look :)

**Comment by fabiangreffrath** [comment]
These two comments were my nits. Sorry, I believed this was obvious. ;)

**Comment by fragglet** [comment]
This has broken several things:

* Top-level INSTALL file included in source tarballs is not generated properly (reads as "These are instructions for how to install and set up LONG_GAME_NAME....") - top-level Makefile.am was not updated
* INSTALL file included in macOS packages has the same problem (pkg/osx/Makefile.am was not updated)
* INSTALL file included in win32 packages has the same problem (pkg/win32/Makefile.am was not updated)

Any chance you can fix these? I took a look but it looks a bit complicated; it might make most sense to centralize all file generation in man/Makefile.am and copy to other places where needed.

**Comment by jmtd** [comment]
Hi @fragglet sorry for breaking stuff! Sure I'll try to fix it up.

**Comment by jmtd** [comment]
I'll push fixes as separate patches to master as I go.

**Comment by jmtd** [comment]
The GNUmakefiles for win32 and osx will now need to be autogen'd. Will that break your/a workflow?

**Comment by fragglet** [comment]
pkg/config.make is autogenerated by configure, and they import from that.

**Comment by jmtd** [comment]
I've raised a PR with changes for those files (#981) because I can't properly test them in my environment: by breaking out the INSTALL stuff as a separate target I could test and am confident the simplecpp invocations are fixed, but just as a matter of good practice it would be nice if someone who has their environment set up for cross-compiling or whatever can make sure they're happy with it?

(I did consider pushing a patch that moved the INSTALL generation to a separate target but the spaces in filenames seemed too risky.)

**Pull request merged**